### PR TITLE
Feature/landing page

### PR DESCRIPTION
**Pull request by BuildNowDpit**
Adbar is included

### CODE REVIEW

**Comment by CimpeanAndreea** [comment]
You have a merge conflict :)

File: lib/appbar/appbar.dart [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+          decoration: BoxDecoration(
+              image: DecorationImage(
+                  image: AssetImage(Constants.blueGradientPath),
+                  fit: BoxFit.fill
+        ))),
+        title: Container(
+              child: Row(
+        children: [
+          Image.asset(
+            Constants.blueLogoNoBgPath,
```
Comment (CimpeanAndreea) [inline lib/appbar/appbar.dart]: Please do not use abbreviations.

File: lib/landingpage/adbar.dart [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+class AdBar extends StatelessWidget {
+  const AdBar({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      width: double.infinity,
+      height: 300,
+      color: Color(0xff176171)
```
Comment (CimpeanAndreea) [inline lib/landingpage/adbar.dart]: Constant for color

File: lib/landingpage/adbar.dart [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import 'package:flutter/material.dart';
+
+import '../constants/constants.dart' as Constants;
+
+class AdBar extends StatelessWidget {
```
Comment (CimpeanAndreea) [inline lib/landingpage/adbar.dart]: "Ad"? Is this a typo?

File: lib/landingpage/category.dart [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  const Category({Key? key, this.categoryName}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        Container(
+          height: double.infinity,
+          width: 2,
+          child: Divider(height: 2, thickness: 28, color: Color(0xffd6d6d6)),
```
Comment (CimpeanAndreea) [inline lib/landingpage/category.dart]: Constant for color

File: lib/landingpage/categorybar.dart [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import 'package:flutter/material.dart';
+
+import './category.dart';
+
+import './categorylist.dart';
+
+class CategoryBar extends StatelessWidget {
+  const CategoryBar({Key? key, screenWidth}) : super(key: key);
+
+  final listOfCategories = const ["Zidărie", "Termoizolații", "Termopane", "Ciment", "Prăfoase", "Acoperiș", "Termopan", "Plexiglas", "Lemn"];
```
Comment (CimpeanAndreea) [inline lib/landingpage/categorybar.dart:12]: @rares2k Regarding making components generic, extracting hard coded values outside and looping, see this example please :D

File: lib/landingpage/landingpage.dart [hunk 6]
```diff
@@ -8,6 +23,22 @@ class LandingPage extends StatefulWidget {
 class _MyWidgetState extends State<LandingPage> {
   @override
   Widget build(BuildContext context) {
-    return Text("Test");
+    return Scaffold(
+      appBar: DefaultAppBar(appBar: AppBar()),
+      body: Container(
+        width: double.infinity,
+        height: double.infinity,
+        color: Color(0xfff2f2f7),
```
Comment (CimpeanAndreea) [inline lib/landingpage/landingpage.dart]: Constant for color

File: lib/landingpage/productslider.dart [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+  const ProductSlider({Key? key, this.productsList}) : super(key: key);
+
+  @override
+  State<ProductSlider> createState() => _MyWidgetState();
+}
+
+class _MyWidgetState extends State<ProductSlider> {
+  final buttonCarouselController = CarouselController();
+
+  static const buttonColor = 0xffe0e0e0;
```
Comment (CimpeanAndreea) [inline lib/landingpage/productslider.dart]: Move it to constants file

File: lib/landingpage/recommendedcategory.dart [hunk 8]
```diff
@@ -0,0 +1,34 @@
+import 'package:flutter/material.dart';
+
+import './productslider.dart';
+
+class ReccomendedCategory extends StatelessWidget {
```
Comment (CimpeanAndreea) [inline lib/landingpage/recommendedcategory.dart]: Typo. Should be "recommended"

File: pubspec.yaml [hunk 9]
```diff
@@ -61,6 +65,8 @@ flutter:
     - assets/bluegradient.png
     - assets/loginpageimage.png
     - assets/bluelogo.png
+    - assets/bluelogo-nobg.png
```
Comment (CimpeanAndreea) [inline pubspec.yaml:69]: Use the same convention for naming, nowhere else is "-" used. Also, don't use abbreviation.

File: lib/landingpage/productslider.dart [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+        child: CarouselSlider.builder(
+          carouselController: buttonCarouselController,
+          options: CarouselOptions(
+              aspectRatio: 2.0, enlargeCenterPage: false, viewportFraction: 1),
+          itemCount: productsCount,
+          itemBuilder: (context, index, realIndex) {
+            final int first = index;
+            final int second = first + 1;
+            return Row(
+              children: [for (var i = 4 * first; i <= 4 * second - 1; i += 1) i]
```
Comment (CimpeanAndreea) [inline lib/landingpage/productslider.dart:59]: I'm not quite sure how this is working, but it doesn't seem to be generic to me. All the data about "distribuitor", "3 review-uri" and so on, would be data that will come from the backend. Try preparing this by making the component accept some sort of parameters and extract the hard coded values outside. Pretty sure that's the approach you used above.

File: lib/landingpage/adbar.dart (hunk 3 above)
Comment (BuildNowDpit) [reply lib/landingpage/adbar.dart]: No, it's not, but this widget could be deleted as it's not used anymore in the app

Comment (CimpeanAndreea) [reply lib/landingpage/adbar.dart]: Perfect, then remove it

File: lib/appbar/appbar.dart (hunk 1 above)
Comment (RobertoWTBW) [reply lib/appbar/appbar.dart]: Resolved

File: lib/landingpage/category.dart (hunk 4 above)
Comment (RobertoWTBW) [reply lib/landingpage/category.dart]: Resolved

File: lib/landingpage/recommendedcategory.dart (hunk 8 above)
Comment (RobertoWTBW) [reply lib/landingpage/recommendedcategory.dart]: Oops :))

File: pubspec.yaml (hunk 9 above)
Comment (RobertoWTBW) [reply pubspec.yaml:69]: Resolved

File: lib/landingpage/productslider.dart (hunk 10 above)
Comment (RobertoWTBW) [reply lib/landingpage/productslider.dart:59]: It's generic, but it's designed to show 4 products at a time

Comment (RobertoWTBW) [reply lib/landingpage/productslider.dart:59]: *maximum, and can work with many more

File: lib/landingpage/landingpage.dart (hunk 6 above)
Comment (RobertoWTBW) [reply lib/landingpage/landingpage.dart]: Resolved

File: lib/landingpage/adbar.dart (hunk 2 above)
Comment (CimpeanAndreea) [inline lib/landingpage/adbar.dart]: Constant

File: lib/landingpage/productslider.dart (hunk 7 above)
Comment (CimpeanAndreea) [reply lib/landingpage/productslider.dart]: The color is still not moved to constants file

**Pull request merged**